### ISSUE TITLE
Support connect-waterline as a session store

### ISSUE THREAD

**Issue by dbkaplun** [balderdashy/sails#2987]
https://github.com/dmarcelino/connect-waterline

**Comment by tjwebb** [comment]
What would need to be done?

**Comment by dbkaplun** [comment]
@dmarcelino maybe you might be able to help out here?

**Comment by dmarcelino** [comment]
I think this makes perfect sense, thanks for the suggestion @dbkaplun. I'm not very familiar with sails code, I mainly use waterline standalone, but I believe the best path would be for sails to provide an instantiated session model, with its own connection, to `waterline-connect`. In terms of code, the following steps would do it:

```js
var ConnectWaterline = require('connect-waterline');

// wherever sails loads collections, add the default session collection
// the connection can refer to the default connection or a specific one for session storing
// (ideally it should be user configurable)
var sessionCollection = Waterline.Collection.extend(_.defaults({ connection: 'connect-waterline' }, ConnectWaterline.defaultModelDefinition))

/**
 * load `sessionCollection` in waterline and initialise waterline... 
 */

// after waterline is initialised add session store to express, passing along the instantiated model object
app.use(session({
  secret: 'foo',
  store: new WaterlineStore({ model: MySessionModel });
}));
```

I guess the place to start in sails would be https://github.com/balderdashy/sails/blob/master/lib/hooks/session/index.js ...

**Comment by tjwebb** [comment]
If it supports the express "connect" interface, what's keeping it from Just
Working?

**Comment by dmarcelino** [comment]
My earlier instructions are to re-use sails waterline instance, if you want something simpler, you can do:

```js
var WaterlineStore = require('connect-waterline');

var options = {
  adapters: {
    'default': require('sails-disk')  // or any other sails adapter
  },
  connections: {
    'connect-waterline': {
      adapter: 'default'
      // specific adapter connection options
      // user: '',
      // password: '',
      // host: 'localhost',
      // database: 'sessions'
    }
  }
})

app.use(session({
  secret: 'foo',
  store: new WaterlineStore(options);
}));
```

This way `connect-waterline` will create its own instance of waterline. You still have to pass the Adapter and any connection details.

**Comment by sailsbot** [comment]
Thanks for posting, @dbkaplun.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
